### DISCUSSION TITLE
'MaskRCNN' object has no attribute 'CLASSES'

### DISCUSSION THREAD

**Discussion by Mrzhans** [obss/sahi#897] [Q&A] [answered]
Hello - I am getting error when trying to run with my custom MaskRCNN model.
I am running this part in my Pycharm.

`
detection_model = MmdetDetectionModel(
    model_path=model_path,
    config_path=config_path,
    confidence_threshold=0.4,
    device="cuda"
)
`

But i am getting error like this.
`Traceback (most recent call last):
  File "D:/py/mmdetection-3.0.0/SAHI_Test.py", line 16, in <module>
    detection_model = AutoDetectionModel.from_pretrained(
  File "D:\Anaconda3\envs\CellDetection\lib\site-packages\sahi\auto_model.py", line 66, in from_pretrained
    return DetectionModel(
  File "D:\Anaconda3\envs\CellDetection\lib\site-packages\sahi\models\base.py", line 67, in __init__
    self.load_model()
  File "D:\Anaconda3\envs\CellDetection\lib\site-packages\sahi\models\mmdet.py", line 39, in load_model
    self.set_model(model)
  File "D:\Anaconda3\envs\CellDetection\lib\site-packages\sahi\models\mmdet.py", line 54, in set_model
    category_mapping = {str(ind): category_name for ind, category_name in enumerate(self.category_names)}
  File "D:\Anaconda3\envs\CellDetection\lib\site-packages\sahi\models\mmdet.py", line 103, in category_names
    if type(self.model.CLASSES) == str:
  File "D:\Anaconda3\envs\CellDetection\lib\site-packages\torch\nn\modules\module.py", line 1207, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'MaskRCNN' object has no attribute 'CLASSES'
`

**Comment by Mrzhans** [comment]
i am sure my model_path is right Because i used it in my own test.py to load the MaskRCNN model.
Here is my test.py:
`config_file = 'configs/mask_rcnn/mask-rcnn_r50_fpn_1x_coco.py'`
```
checkpoint_file = 'work_dirs/mask-rcnn_r50_fpn_1x_coco/best_coco_bbox_mAP_epoch_32.pth'
model = init_detector(config_file, checkpoint_file, device='cuda:0')
visualizer = VISUALIZERS.build(model.cfg.visualizer)
visualizer.dataset_meta = model.dataset_meta
img = mmcv.imread('C:/Users/Yjt/Desktop/CellDetection/CellDatasets/images/HCJI_12-22.png')
result = inference_detector(model, img)
img = mmcv.imread(img)
img = mmcv.imconvert(img, 'bgr', 'rgb')

visualizer.add_datasample(
    'result',
    img,
    data_sample=result,
    draw_gt=False,
    show=True)
```
Then,show the result image.

When i run this code:
```
model_path = 'work_dirs/mask-rcnn_r50_fpn_1x_coco/best_coco_bbox_mAP_epoch_32.pth'
config_path = 'configs/mask_rcnn/mask-rcnn_r50_fpn_1x_coco.py'

detection_model = MmdetDetectionModel(
    model_path=model_path,
    config_path=config_path,
    confidence_threshold=0.4,
    device="cuda"
)
```
I am getting error 'MaskRCNN' object has no attribute 'CLASSES'

**Comment by fcakyon** [comment, marked as answer]
What is you mmdet version? v3 is currently not supported.

**Reply by Mrzhans** [reply]
omg，my version is 3.00!!!
